salalib/spacepix.cpp

```diff
@@ -352,16 +324,15 @@ void SpacePixel::construct(const SpacePixel& spacepixel)
    m_newline = true;
 
    if (!m_rows || !m_cols) {
-      m_pixel_lines = NULL;
-      m_display_lines = NULL;
+      m_pixel_lines.clear();
+      m_display_lines.clear();
       return;
    }
 
-   m_pixel_lines = new pvecint *[m_cols];
+   m_pixel_lines = std::vector<std::vector<int> >(size_t(m_cols*m_rows));

```

Comment:
Rolling out a matrix into a vector of size row x cols is probably the right way to do it - but we should really provide a helper class to do that. Having to calculate the index inside your business logic is hard to read and error prone...

Counter-intuitively, here you have a vector<int> at each pixel, so it took me the best part of 5 minutes to figure out what is going on in this one line.